### PR TITLE
Contralib sample

### PR DESCRIPTION
**Pull request by joejstuart**
This provides an example Jenkinsfile of creating a rpm build from a PR and a build from a koji task. It also provides the s2i for the required containers.

### CODE REVIEW

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild [hunk 1]
```diff
@@ -0,0 +1,86 @@
+CANNED_CI_MESSAGE = '{"build_id":1104742,"old":0,"name":"krb5","task_id":28114680,"attribute":"state","request":["git+https://src.fedoraproject.org/rpms/krb5.git#40a05d03472a1bcc7aa57f32273357732b2fe895","f28-candidate",{}],"instance":"primary","epoch":null,"version":"1.16.1","owner":"rharwood","new":1,"release":"10.fc28","topic":"org.fedoraproject.prod.buildsys.build.state.change"}'
+
+def libraries = ['contra-lib': ['jmsMessages', 'https://github.com/joejstuart/contra-lib.git']]
```
Comment (johnbieren) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild]: /me notes this points at a fork

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+                        ]
+                )
+        ]
+)
+
+containers = [
+        'rpmbuild':
+        [
+                tag: env.RPMBUILD_TAG
```
Comment (johnbieren) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild]: I don't think any of these tags are defined in env

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+                        ]
+                ),
+        ]
+)
+
+containers = [
+        'rpmbuild':
+        [
+                tag: env.RPMBUILD_TAG
```
Comment (johnbieren) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild]: Same as above

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                buildDiscarder(logRotator(artifactDaysToKeepStr: '30', artifactNumToKeepStr: '', daysToKeepStr: '90', numToKeepStr: '')),
+                disableConcurrentBuilds(),
+                parameters(
+                        [
+                                string(defaultValue: "${env.ghprbActualCommit}", description: '', name: 'ghprbActualCommit'),
+                                string(defaultValue: "${env.ghprbGhRepository}", description: '', name: 'ghprbGhRepository'),
+                                string(defaultValue: '', description: '', name: 'ghprbPullId'),
+                                string(defaultValue: '', description: '', name: 'ghprbPullAuthorLogin'),
+                                string(defaultValue: 'continuous-infra', description: 'Project namespace for Openshift operations', name: 'OPENSHIFT_NAMESPACE'),
+                                string(defaultValue: '172.30.1.1:5000', description: 'The internal docker registry url', name: 'DOCKER_REPO_URL'),
```
Comment (johnbieren) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild]: Maybe this becomes env.DOCKER_REPO_URL like the other Jenkinsfile in this PR?

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                                string(defaultValue: '', description: '', name: 'ghprbPullId'),
+                                string(defaultValue: '', description: '', name: 'ghprbPullAuthorLogin'),
+                                string(defaultValue: "${env.OPENSHIFT_NAMESPACE}", description: 'Project namespace for Openshift operations', name: 'OPENSHIFT_NAMESPACE'),
+                                string(defaultValue: "${env.DOCKER_REPO_URL}", description: 'The internal docker registry url', name: 'DOCKER_REPO_URL'),
+                                string(defaultValue: 'stable', description: 'The rpmbuild image tag', name: 'RPMBUILD_TAG'),
+                                string(defaultValue: 'stable', description: 'The cloud-image-compose image tag', name: 'CLOUD_IMAGE_COMPOSE_TAG'),
+                                string(defaultValue: 'stable', description: 'The singlehost-test image tag', name: 'SINGLEHOST_TEST_TAG'),
+                                string(defaultValue: CANNED_CI_MESSAGE, description: 'The ci message to build from', name: 'CI_MESSAGE')
```
Comment (johnbieren) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild]: In the other file, this was ${CANNED_CI_MESSAGE}, not sure if one works and the other doesn't, or if both do, but probably best to be consistent

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild (hunk 2 above)
Comment (joejstuart) [reply config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfileKojiTaskBuild]: Noted. Changed to latest. thanks

File: config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild [hunk 6]
```diff
@@ -0,0 +1,90 @@
+/*
+This requires a keytab to run the rpm-build step.
```
Comment (arilivigni) [inline config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild]: Do you need a keytab from Fedora infra or can you also use your own kerb creds?

Comment (joejstuart) [reply config/pipelines/Jenkinsfile-contra-lib-sample/JenkinsfilePRBuild]: You can use your own. I updated the docs.

**Pull request merged**